### DISCUSSION TITLE
osu! no longer running in Exclusive Fullscreen when using newer AMD drivers

### DISCUSSION THREAD

**Discussion by SupDos** [ppy/osu#29647] [Q&A]
_For context, from my testing lazer will only ever run in exclusive fullscreen with the OpenGL renderer - any other renderers run in borderless/fullscreen optimizations mode_

With driver version 24.5.1, osu! runs in Exclusive fullscreen, as confirmed by Presentmon showing **Hardware: Legacy Flip**

![OLD_OpenGL](https://github.com/user-attachments/assets/7308d1e0-f7bd-4cf0-a36c-993eebc8124a)

However, with the next driver update after this one, 24.6.1, 24.7.1 and 24.8.1 (which seems to have been released while I was making this) it will no longer run in exclusive, and will show the volume controls even when the game is focused. Presentmon details this as **Hardware: Independent Flip**

![MID_OpenGL](https://github.com/user-attachments/assets/1d4ce0e0-968e-4a16-bb52-921a071ff92a)

![NEW_OpenGl](https://github.com/user-attachments/assets/12d1e2e9-d57d-4410-9c16-07bb193094c2)

I have also tried this with the SDL3 environment variable just in case it helped, but still see the same effects.

---

While I am sure this is an AMD issue, I'm wondering if lazer's exclusive fullscreen method is off somehow, as a different game I play, Overwatch 2, can still enter exclusive fullscreen on the newer drivers (note the **Hardware: Legacy Flip**)

![NEW_Ow](https://github.com/user-attachments/assets/30d03790-cc71-49f3-af42-fb181c122ac5)

Posting this in case anyone has a similar issue - for now I can downgrade to 24.5.1 and fullscreen works fine, but  if anyone knows how to fix it apart from AMD fixing it themselves (very unlikely) it would be appreciated

**Comment by smoogipoo** [comment]
Reporting to AMD would be good. Can you also test whether changing the fullscreen optimizations setting in right click -> properties does anything? Historically that setting only had an effect on D3D from my recollection, but maybe it's changed. You may need to apply it on the actual executable (`%LOCALAPPDATA%/osu/{version}`), so test that also.

**Reply by SupDos** [reply]
> fullscreen optimizations setting in right click -> properties

have tried this, both changing it in the first executable (`AppData/Local/osu!.exe`) and the actual executable in the version folder, to no avail sadly.

I got a reply from someone at AMD on their [subreddit](https://old.reddit.com/r/Amd/comments/1f45m7v/amd_software_adrenalin_edition_2481_release_notes/lkk1z4g/?context=10000) who said they'd reported it internally a few months ago and would be checking up on it, so hopefully I get some news there - otherwise I'll report it through a different channel of theirs.

**Reply by Vikingtons** [reply]
@smoogipoo I don't suppose either you or ppy could join us in the Vanguard community Discord if you get a chance? @SupDos I may shoot you an invite to verify this issue on a later driver branch

**Reply by smoogipoo** [reply]
Hey @Vikingtons, sorry I missed your Reddit message... I've joined the server and would be glad to help out any way I can, let me know what you need.

**Comment by smoogipoo** [comment]
@SupDos Can you download https://www.capframex.com and capture two frames, one with the old driver (in exclusive fullscreen), and one with the new driver?

**Reply by SupDos** [reply]
24.5.1
[CapFrameX-osu!.exe-2024-09-01T194956.json](https://github.com/user-attachments/files/16830002/CapFrameX-osu.exe-2024-09-01T194956.json)

24.8.1
[CapFrameX-osu!.exe-2024-09-01T195947.json](https://github.com/user-attachments/files/16830005/CapFrameX-osu.exe-2024-09-01T195947.json)

In both of these I'm just sitting on the main menu, not doing anything apart from pressing the capture button - also made sure that I was in exclusive fullscreen in 24.5.1 (no volume controls would show when changing volume as expected) while in 24.8.1 volume controls do show, without changing any settings in between driver upgrades.

**Reply by smoogipoo** [reply]
Are you running unlimited fps when testing this?

**Reply by smoogipoo** [reply]
Can you also take a captures in windowed + borderless modes on the new driver for reference?

**Comment by smoogipoo** [comment]
Please upload logs from the game (`Options -> Export Logs`).

**Reply by SupDos** [reply]
Exported them twice to keep it organized (though the last export probably has the previous logs as well):
24.5.1
[compressed-logs-24-5-1.zip](https://github.com/user-attachments/files/16830008/compressed-logs-24-5-1.zip)

24.8.1
[compressed-logs-24-8-1.zip](https://github.com/user-attachments/files/16830009/compressed-logs-24-8-1.zip)

**Comment by smoogipoo** [comment]
By any chance, do you have a phone with a slo-mo recording mode? The [captures](https://github.com/ppy/osu/discussions/29647#discussioncomment-10513826) above look identical in both cases so I'm trying to quantify the input lag in some way.

**Reply by SupDos** [reply]
Yep I do, I've recorded both in 240fps slo-mo and personally I can't see any difference between hand movement and on screen movement. I can only go off by what I feel which is definitely not helpful :(

**Comment by smoogipoo** [comment]
In the old exclusive fullscreen, especially when running unlimited, you should see some tearing. It'll be more obvious in taiko than the other modes. Do you see tearing at all in the new driver?

**Reply by SupDos** [reply]
Yep, I do see tearing with the new driver, though I notice it more at 2x compared to unlimited - but it's definitely there.

**Comment by smoogipoo** [comment]
Can you also try out the D3D11 renderer on the new driver in the following modes:
- Borderless fullscreen @ unlimited
- Fullscreen @ unlimited

And see if your perceived latency is lower. It will use iFlip, but we've made what I believe is our best attempt at lowering latency as far as D3D11 can take us. Maybe provide PerfMon logs or CapFrameX captures too.

**Reply by SupDos** [reply]
I'll try D3D11 using the new driver for the next few days and report back if necessary.

Originally I only reported this because it was reproducible by downgrading and upgrading the driver every time, so I thought it must have been a side effect of something else (and also just not wanting to deal with fullscreen optimization jank). If it's not as easy as flipping a switch to bring back exclusive fullscreen on the newer drivers, and seeing all the captures I've made showing practically no difference between Legacy Flip and Independent Flip, then I will just have to get used to it.

Adding captures here as well in the meantime.

24.8.1, D3D11, Unlimited, Borderless (for some reason borderless locks my refresh rate in d3d11 even when using unlimited, so not sure if this is very useful)
[CapFrameX-osu!.exe-2024-09-02T133321.json](https://github.com/user-attachments/files/16849831/CapFrameX-osu.exe-2024-09-02T133321.json)

24.8.1, D3D11, Unlimited, Fullscreen
[CapFrameX-osu!.exe-2024-09-02T133510.json](https://github.com/user-attachments/files/16849834/CapFrameX-osu.exe-2024-09-02T133510.json)

**Reply by smoogipoo** [reply]
> borderless locks my refresh rate in d3d11

That's intended and one of the things I wanted to look at, actually. Although the fullscreen capture there appears to have lower latency than OGLP (both versions) by numbers alone, they're both on the same magnitude so I'd say AMD is likely doing things correctly there.
